source/extensions/tracers/xray/tracer.cc

```diff
@@ -76,17 +77,28 @@ void Span::finishSpan() {
   s.set_end_time(
       time_point_cast<SecondsWithFraction>(time_source_.systemTime()).time_since_epoch().count());
   s.set_parent_id(parentId());
-  using KeyValue = Protobuf::Map<std::string, std::string>::value_type;
-  for (const auto& item : custom_annotations_) {
-    s.mutable_annotations()->insert(KeyValue{item.first, item.second});
+
+  // HTTP annotations
+  using StructField = Protobuf::MapPair<std::string, ProtobufWkt::Value>;
+  daemon::Segment_http_annotations* http = s.mutable_http();
+
+  ProtobufWkt::Struct* request = http->mutable_request();
+  auto request_fields = request->mutable_fields();
```

Comment:
nit: `auto*` instead of `auto` when the deduced type is a pointer.

Reply:
fixed.